src/app/create-guild/_components/CreateGuildImageUploader.tsx

```diff
@@ -0,0 +1,119 @@
+import { Button } from "@/components/ui/Button"
+import { FormControl, FormErrorMessage, FormItem } from "@/components/ui/Form"
+import { Image, Spinner } from "@phosphor-icons/react/dist/ssr"
+import {
+  getWidthAndHeightFromFile,
+  imageDimensionsValidator,
+} from "components/create-guild/IconSelector/utils"
+import useDropzone, { ERROR_MESSAGES } from "hooks/useDropzone"
+import usePinata from "hooks/usePinata"
+import { useState } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
+import { CreateGuildFormType } from "../types"
+
+const MIN_WIDTH = 512
+const MIN_HEIGHT = 512
+
+/**
+ * This is a pretty specific component right now, but we should generalise it once we start using it in other places too (e.g. on the create profile page)
```

Comment:
I wouldn't like to implement unnecessary logic, which we possibly won't actually use in the future, that's why I decided to go with a "specific" component here. ;)